.github/workflows/ci.yml

```diff
@@ -358,6 +358,59 @@ jobs:
         CXX_STANDARD: '20'
       run: ./ci/do_ci.sh cmake.c++20.stl.test
 
+  cmake_test_cxx23_gcc:
+    name: CMake C++23 test(GCC)
+    runs-on: ubuntu-20.04
```

Comment:
why not to use `ubuntu-latest`?

Reply:
Copy/paste. I think from the c++20 version.

I'll go with whatever you suggest. (Fell free to send an edit. I'm OOO right now.)

Reply:
Using a mix of different OS releases improves coverage. To get a recent C++23 compiler, a more recent version of the distribution is most likely needed, so using ubuntu-latest seems better.

Reply:
[Pushed.](https://github.com/open-telemetry/opentelemetry-cpp/pull/2496/commits/8c060284d52477015ce05e7cafca2bdc7d96d2c9)

Reply:
Thanks.

https://github.com/actions/runner-images/blob/ubuntu22/20240116.3/images/ubuntu/Ubuntu2204-Readme.md

has clang-15 and gcc-13

https://github.com/actions/runner-images/blob/ubuntu22/20240116.3/images/ubuntu/Ubuntu2004-Readme.md

has clang-12 and gcc-10

so this should be better.

Resolved.